### PR TITLE
Rename wagtail.contrib.wagtailstyleguide to wagtail.contrib.styleguide

### PR DESCRIPTION
**Pull request by iKevinY**
I decided to try my hand at renaming a module as described in #1504. I'm new to Wagtail's codebase, so I tried to follow @kaedroho's existing module-renaming PRs as closely as possible — hopefully I didn't overlook anything too significant!

### CODE REVIEW

File: wagtail/contrib/styleguide/wagtail_hooks.py [hunk 1]
```diff
@@ -11,10 +11,10 @@
 @hooks.register('register_admin_urls')
 def register_admin_urls():
     return [
-        url(r'^styleguide/$', views.index, name='wagtailstyleguide'),
+        url(r'^styleguide/$', views.index, name='styleguide'),
```
Comment (kaedroho) [inline wagtail/contrib/styleguide/wagtail_hooks.py]: The URL should remain ``wagtailstyleguide``

**Comment by kaedroho** [comment]
Thanks, except for my comment above, this looks good!

I won't merge this yet as I'd like to get all the renames for the contrib modules merged around the same time, and my other pull requests haven't been reviewed yet...

**Comment by tomdyson** [comment]
Thanks a lot, @iKevinY

**Comment by kaedroho** [comment]
Just noticed, ``wagtail/contrib/wagtailstyleguide/__init__.py`` needs to contain: ``default_app_config = 'wagtail.contrib.wagtailstyleguide.apps.WagtailStyleGuideAppConfig'``.

File: wagtail/contrib/styleguide/wagtail_hooks.py (hunk 1 above)
Comment (iKevinY) [inline wagtail/contrib/styleguide/wagtail_hooks.py]: Whoops; thanks for catching this!

Comment (kaedroho) [inline wagtail/contrib/styleguide/wagtail_hooks.py]: Yep, look for all the calls to ``reverse('styleguide')`` and change them there as well

Comment (iKevinY) [inline wagtail/contrib/styleguide/wagtail_hooks.py]: Ah, that makes sense.

**Comment by iKevinY** [comment]
@tomdyson No worries!

@kaedroho Thanks for the review! I've fixed up the commits and rebased.

**Comment by iKevinY** [comment]
Noticed that some merge conflicts were introduced, so I've rebased on top of `master` and pushed.

**Comment by iKevinY** [comment]
Rebased again to fix merge conflicts introduced by 84563247f691917697edd57c27950c2c7ee8191b. The build is failing due to JavaScript code style check, but it seems #1813 will resolve this.

**Comment by timheap** [comment]
You can also rebase on top of #1813, which will allow this PR to pass its checks. You will then not have to do anything when #1813 is merged.

```
git remote add takeflight https://github.com/takeflight/wagtail.git
git fetch takeflight
git checkout wagtailstyleguide-rename
git rebase takeflight/fix/javascript-style-error
git push origin -f
```

**Comment by iKevinY** [comment]
@timheap Good idea; I forgot about the fact that you could fetch PRs as if they were actual branches.

**Comment by SalahAdDin** [comment]
:+1:

**Comment by kaedroho** [comment]
Sorry to bump this again. We still haven't decided the approach to renaming these modules while maintaining backwards compatibility.

**Comment by iKevinY** [comment]
No problem, I'll keep the PR open anyways for when that decision is made.

File: wagtail/contrib/styleguide/apps.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from django.apps import AppConfig
+
+
+class WagtailStyleGuideAppConfig(AppConfig):
+    name = 'wagtail.contrib.styleguide'
+    label = 'wagtailstyleguide'
```
Comment (BertrandBordage) [inline wagtail/contrib/styleguide/apps.py]: Should we keep this label? It feels useless.

**Comment by BertrandBordage** [comment]
In fact, this pull request could be closed in favor of the broader solution adopted in #2089.

**Comment by iKevinY** [comment]
@BertrandBordage Ah, it's been so long since I made this PR! It probably makes sense to close this in favour of #2089.

**Comment by BertrandBordage** [comment]
Closing in favour of #2089.

**Pull request closed without merging**